CMakeLists.txt

```diff
@@ -85,6 +89,26 @@ add_subdirectory(source)
 add_subdirectory(examples)
 add_subdirectory(test)
 
+find_program(CLANG_FORMAT NAMES clang-format-10 clang-format-10.0 clang-format)
+
+set(CLANG_FORMAT_REQUIRED "10.0")
+
+if(CLANG_FORMAT)
+	get_program_version_major_minor(${CLANG_FORMAT} CLANG_FORMAT_VERSION)
+	message(STATUS "Found clang-format: ${CLANG_FORMAT} (version: ${CLANG_FORMAT_VERSION})")
+endif()
+
+# Check if the correct version of clang-format is available
+if(FORMAT_CPP_STYLE)
+	if(CLANG_FORMAT)
+		if(NOT (CLANG_FORMAT_VERSION VERSION_EQUAL CLANG_FORMAT_REQUIRED))
```

Comment:
Unfortunately, the formatting might differ between versions, so we need to require a specific one. And this is a pain when upgrading a distribution on a dev machine - there's usually only one version of clang-format available in the package manager.

Some projects use a separate docker image with a specific clang-format version for that, e.g.: https://github.com/intel/CacheLib/blob/main/.github/workflows/clang-format-check.yml#L16. The drawback is that you cannot clang-format inside a different docker image, but it's not a big problem IMO.

Reply:
Well, that's a bummer. I just checked, and on Arch clang-format with the entire `clang` package, so there's no easy way to downgrade...
Let's do the docker image and a separate workflow like your cachelib example.

Reply:
There is a question about whether we want to integrate this with CMake. Do we want a target that would run the formatting using the docker image? 

Also, With this approach, we would have a dev dependency on docker. Probably we would also need to host our docker image (on gh registry probably) since the image used by CacheLib is hosted here: https://github.com/jidicula/clang-format-action/pkgs/container/clang-format and I don't think we have any guarantees that those images will be always available.

Reply:
hmm... integrating with CMake isn't a bad idea. But I think the target should first try to run a local `clang-format` if it exists in the correct version.
And yes, we should definitely have create own docker image.

Reply:
So, probably the existing integration can stay, and we can just fall back to docker in case there is no clang format installed (in the required version). The docker integration could even be done in a separate PR.